### PR TITLE
(21.lts.1+) Add on-device tests

### PR DESCRIPTION
**Pull request by iamjazzar**

### CODE REVIEW

File: tools/on_device_tests_gateway.proto [hunk 1]
```diff
@@ -0,0 +1,54 @@
+// Copyright 2022 The Cobalt Authors. All Rights Reserved.
```
Comment (isarkis) [inline tools/on_device_tests_gateway.proto:1]: Do we have CL to bring tools/on_device_tests* files to cobalt_src on GoB?

File: .github/config/raspi-2.json [hunk 2]
```diff
@@ -1,6 +1,5 @@
 {
   "docker_service": "build-raspi",
-  "on_device_test": "true",
```
[the remark this answers is not in the document]
Comment (iamjazzar) [reply .github/config/raspi-2.json:3]: That's for Evergreen raspi loader, Here's the pure raspi-2 https://cobalt-buildbot.corp.goog/builders/raspi-2_nightly_21.lts.1%2B/builds/693

[the remark this answers is not in the document]
Comment (iamjazzar) [reply .github/config/raspi-2.json:3]: That's for Evergreen raspi loader, Here's the pure raspi-2  go/rapi-2-pure-test

[the remark this answers is not in the document]
Comment (isarkis) [reply .github/config/raspi-2.json:3]: You are right!

File: .github/config/evergreen-arm-hardfp.json [hunk 3]
```diff
@@ -1,7 +1,13 @@
 {
   "docker_service": "build-raspi",
   "bootloader": "true",
-  "on_device_test": "true",
+  "on_device_test": {
+    "enabled": true,
+    "tests": [
+      "unit_test",
```
[the remark this answers is not in the document]
Comment (iamjazzar) [reply .github/config/evergreen-arm-hardfp.json:7]: That's not for lts, here's the 21 lts version http://go/evergreen-pure-test

File: tools/on_device_tests_gateway.proto (hunk 1 above)
Comment (iamjazzar) [reply tools/on_device_tests_gateway.proto:1]: Not sure, I haven't done that

File: .github/config/evergreen-arm-hardfp.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (isarkis) [reply .github/config/evergreen-arm-hardfp.json:7]: You are right!

File: tools/on_device_tests_gateway.proto (hunk 1 above)
Comment (isarkis) [reply tools/on_device_tests_gateway.proto:1]: Let's do that before submitting this PR.  I am not sure if 20.lts.1+ branch in GoB has precommit, but if it does we might want to have that CL go through precommit.  

FYI @andrewsavage1

Comment (iamjazzar) [reply tools/on_device_tests_gateway.proto:1]: I created a ticket for tools/on_device_tests on GoB. I will be working on it once all the legacy branches changes are ready to be submitted.

**Review by isarkis — APPROVED** [review]

**Pull request merged**